jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/JKubeException.java

```diff
@@ -19,4 +19,32 @@ public JKubeException(String message) {
     super(message);
   }
 
+  public JKubeException(String message, Throwable cause) {
+    super(message, cause);
+  }
+
+  /**
+   * Convert exception into an instance of JKubeException
+   *
+   * @param message message of exception
+   * @param cause actual cause of exception
+   * @return {@link RuntimeException} wrapping actual cause
+   */
+  public static RuntimeException launderThrowable(String message, Throwable cause) {
+    RuntimeException processed = processCause(cause);
+    if (processed != null)
+      return processed;
+    throw new JKubeException(message, cause);
+  }
+
+  private static RuntimeException processCause(Throwable cause) {
+    if (cause instanceof RuntimeException) {
+      return ((RuntimeException) cause);
+    } else if (cause instanceof Error) {
+      throw ((Error) cause);
+    } else if (cause instanceof InterruptedException) {
+      Thread.currentThread().interrupt();
+    }
+    return null;
+  }
```

Comment:
This method is not reused elsewhere, there's no point in extracting it and then doing a redundant if-else check in launderThrowable